### PR TITLE
Resolve TS errors: TreeMapper

### PR DESCRIPTION
**Pull request by prachigarg19**
This PR resolves typescript errors

### CODE REVIEW

**Comment by mohitb35** [comment]
@prachigarg19 Please resolve conflicts. Is this otherwise complete, or is anything left?

**Comment by prachigarg19** [comment]
> @prachigarg19 Please resolve conflicts. Is this otherwise complete, or is anything left?

It's complete

**Review by mohitb35 — COMMENTED** [review]
test

**Review by mohitb35 — CHANGES_REQUESTED** [review]
@prachigarg19 Please go through my feedback

File: src/features/user/TreeMapper/components/TreeMapperList.tsx [hunk 1]
```diff
@@ -4,15 +4,20 @@ import TransactionsNotFound from '../../../../../public/assets/images/icons/Tran
...
+  PlantLocation as PlantLocationType,
+} from '../../../common/types/plantLocation';
+import { Links } from '../../../common/types/payments';
 
 interface Props {
   selectedLocation: string;
   setselectedLocation: Function;
-  plantLocations: Object;
+  plantLocations: PlantLocationType[] | SamplePlantLocation[] | null;
   isDataLoading: boolean;
```
[the remark this answers is not in the document]
Comment (prachigarg19) [reply src/features/user/TreeMapper/components/TreeMapperList.tsx:17]: Yes, properties match with `PlantLocation` in `plantLocation.d.ts`.

File: src/features/user/TreeMapper/index.tsx [hunk 2]
```diff
@@ -89,8 +96,8 @@ function TreeMapper(): ReactElement {
             for (const itr in plantLocations) {
               if (Object.prototype.hasOwnProperty.call(plantLocations, itr)) {
                 const location = plantLocations[itr];
-                if (location.type === 'multi') {
-                  plantLocations[itr].sampleTrees = [];
+                if (location.type === 'multi' && 'sampleTrees' in location) {
+                  location.sampleTrees = [];
```
[the remark this answers is not in the document]
Comment (mohitb35) [reply src/features/user/TreeMapper/index.tsx]: The `&& 'sampleTrees' in location` condition was removed as it made the code in the `if` block unreachable.

File: src/features/user/TreeMapper/Import/components/SampleTrees.tsx [hunk 3]
```diff
@@ -60,17 +63,18 @@ export default function SampleTrees({
...
-        const csv = event.target.result;
+      reader.onload = (event) => {
+        const csv = event.target?.result;
         Papa.parse(csv, {
           header: true,
-          complete: (results: any) => {
-            const sampleTrees = results.data.map((sampleTree: SampleTree) => {
+          complete: (results) => {
+            const resultsData: SampleTree[] = results.data;
+            const sampleTrees: SampleTree[] = resultsData.map((sampleTree) => {
```
[the remark this answers is not in the document]
Comment (mohitb35) [reply src/features/user/TreeMapper/Import/components/SampleTrees.tsx]: Have created an issue #1872 relating to the CSV validation and the pending warning `Type 'unknown[]' is not assignable to type 'SampleTree[]'` seen for `resultsData`.

**Review by mohitb35 — APPROVED** [review]
LGTM now.

**Pull request merged**